### PR TITLE
Revbump retext: Update PyQt5 deps for >=5.12.

### PR DESCRIPTION
**Pull request by holgersson32644**

### CODE REVIEW

File: app-editors/retext/retext-7.0.4-r1.ebuild [hunk 1]
```diff
@@ -0,0 +1,73 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+# no pypy{,3} support as PyQt5 does not support it at 2019-04-13
+# https://bitbucket.org/pypy/compatibility/wiki/Home#!gui-library-bindings
+PYTHON_COMPAT=( python3_{5,6,7} )
+
+inherit distutils-r1 gnome2-utils virtualx xdg-utils
```
Comment (juippis) [inline app-editors/retext/retext-7.0.4-r1.ebuild]: gnome2_icon_cache update doesn't work with EAPI-7 anymore. Use xdg_icon_cache_update from xdg-utils instead.

File: app-editors/retext/retext-7.0.4-r1.ebuild [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+	dev-python/PyQt5[gui,network,printsupport,widgets,${PYTHON_USEDEP}]
+	|| (
+		( >=dev-python/PyQt5-5.12 dev-python/PyQtWebEngine )
+		dev-python/PyQt5[webengine]
+		)
+	spell? ( dev-python/pyenchant[${PYTHON_USEDEP}] )
+"
+DEPEND="
+	${RDEPEND}
+	dev-python/setuptools[${PYTHON_USEDEP}]
```
Comment (juippis) [inline app-editors/retext/retext-7.0.4-r1.ebuild]: With EAPI-7 this is a BDEPEND. distutils-r1 will put it there for you, but doesn't hurt to show it's being used.

**Comment by holgersson32644** [comment]
Hi juippis,

thanks for the review.

> +inherit distutils-r1 gnome2-utils virtualx xdg-utils
> 
> gnome2_icon_cache update doesn't work with EAPI-7 anymore. Use
> xdg_icon_cache_update from xdg-utils instead.

Fixed and finally written down to my checklist for future bumps.

> +DEPEND="
> +	${RDEPEND}
> +	dev-python/setuptools[${PYTHON_USEDEP}]
> 
> With EAPI-7 this is a BDEPEND. distutils-r1 will put it there for you,
> but doesn't hurt to show it's being used.

Fixed.

I also screwed up the removal commit a bit which is fixed now.

File: app-editors/retext/retext-7.0.4-r1.ebuild [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+RDEPEND="
+	dev-python/chardet[${PYTHON_USEDEP}]
+	dev-python/docutils[${PYTHON_USEDEP}]
+	dev-python/markdown[${PYTHON_USEDEP}]
+	dev-python/markups[${PYTHON_USEDEP}]
+	dev-python/pygments[${PYTHON_USEDEP}]
+	dev-python/python-markdown-math[${PYTHON_USEDEP}]
+	dev-python/PyQt5[gui,network,printsupport,widgets,${PYTHON_USEDEP}]
+	|| (
+		( >=dev-python/PyQt5-5.12 dev-python/PyQtWebEngine )
```
Comment (mgorny) [inline app-editors/retext/retext-7.0.4-r1.ebuild]: What about PYTHON_USEDEP?

File: app-editors/retext/retext-7.0.4-r1.ebuild [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+		dev-python/PyQt5[webengine]
+		)
+	spell? ( dev-python/pyenchant[${PYTHON_USEDEP}] )
+"
+DEPEND="
+	${RDEPEND}
+	test? ( x11-base/xorg-server[xvfb] )
```
Comment (mgorny) [inline app-editors/retext/retext-7.0.4-r1.ebuild]: Why do you need this explicitly? virtualx should take care of it.

File: app-editors/retext/retext-7.0.4-r1.ebuild [hunk 5]
```diff
@@ -36,13 +36,18 @@ RDEPEND="
 	dev-python/markups[${PYTHON_USEDEP}]
 	dev-python/pygments[${PYTHON_USEDEP}]
 	dev-python/python-markdown-math[${PYTHON_USEDEP}]
-	dev-python/PyQt5[gui,network,printsupport,webengine,widgets,${PYTHON_USEDEP}]
+	dev-python/PyQt5[gui,network,printsupport,widgets,${PYTHON_USEDEP}]
```
Comment (a17r) [inline app-editors/retext/retext-7.0.4-r1.ebuild]: This block can be simplified though, right?

* Either more descriptive:
```
|| (
	(
		dev-python/PyQtWebEngine[${PYTHON_USEDEP}]
		>=dev-python/PyQt5-5.12[gui,network,printsupport,widgets,${PYTHON_USEDEP}]
	)
	dev-python/PyQt5[gui,network,printsupport,webengine,widgets,${PYTHON_USEDEP}]
)
```
* Or compact:
```
dev-python/PyQt5[gui,network,printsupport,widgets,${PYTHON_USEDEP}]
|| (
	dev-python/PyQtWebEngine[${PYTHON_USEDEP}]
	>=dev-python/PyQt5-5.12[webengine,${PYTHON_USEDEP}]
)
```

Please also note: One line per dependency.

**Comment by a17r** [comment]
ping ;)

**Pull request closed without merging**